src/sciline/pipeline.py

```diff
@@ -511,7 +527,9 @@ def build(
             if (optional_arg := get_optional(tp)) is not None:
                 try:
                     optional_subgraph = self.build(
-                        optional_arg, search_param_tables=search_param_tables
+                        optional_arg,
+                        search_param_tables=search_param_tables,
+                        handler=HandleAsBuildTimeException(),
```

Comment:
Why does this use a build time error instead of forwarding the handler?

Reply:
This is handling an "optional" branch. I thought if a requirement is optional we should prune the entire branch, as before?

Reply:
Makes sense. But we could also show optional requirements in the viz. But that should be a separate PR. And it's not clear to me whether we'd want to show only the immediate requirement or all transitive requirements, too.

Reply:
Currently it shows the immediate requirements.